--- src/MainWindow.cpp
+++ src/MainWindow.cpp
@@ -166,7 +166,7 @@
     vfsModel = new QStandardItemModel(this);
     vfsExplorer->setModel(vfsModel);
     vfsExplorer->setViewMode(QListView::IconMode);
-    vfsExplorer->setMovement(QListView::Static);
+    vfsExplorer->setDragDropOverwriteMode(true);
     vfsExplorer->setAcceptDrops(true);
     vfsExplorer->setDragEnabled(true);
     vfsExplorer->setDragDropMode(QAbstractItemView::DragDrop);
