--- src/MainWindow.cpp
+++ src/MainWindow.cpp
@@ -248,9 +248,10 @@
         if (title == "*New Item*") title = "Unsaved Draft";
         else title = "Draft of: " + title;

+        QString textToSave = (mainContentStack->currentWidget() == docContainer) ? documentEditorView->toPlainText() : noteEditorView->toPlainText();
         if (currentAutoDraftId == 0) {
-            currentAutoDraftId = currentDb->addDraft(0, title, documentEditorView->hasFocus() ? documentEditorView->toPlainText() : noteEditorView->toPlainText());
+            currentAutoDraftId = currentDb->addDraft(0, title, textToSave);
         } else {
-            currentDb->updateDraft(currentAutoDraftId, title, documentEditorView->hasFocus() ? documentEditorView->toPlainText() : noteEditorView->toPlainText());
+            currentDb->updateDraft(currentAutoDraftId, title, textToSave);
         }
     });
