--- src/MainWindow.cpp
+++ src/MainWindow.cpp
@@ -3651,7 +3651,7 @@
             sourceView = qobject_cast<QAbstractItemView*>(sourceWidget->parentWidget());
         }

-        if (dropEvent->mimeData()->hasUrls()) {
+        if (dropEvent->mimeData()->hasUrls() && sourceView != openBooksTree && sourceView != vfsExplorer) {
             // external files
             QList<QUrl> urls = dropEvent->mimeData()->urls();
             if (!urls.isEmpty() && urls.first().isLocalFile()) {
